[api] fix behaviour on project meta change of scmsync project

adrianschroeter opened the pull request:
We must only decline requests using the project as target, but not as source.

Otherwise any random meta change leads to revoke requests when a project meta change happens.

adrianschroeter committed:
[api] fix behaviour on project meta change of scmsync project

We must only decline requests using the project as target,
but not as source.

Otherwise any random meta change leads to revoke requests when
a project meta change happens.

adrianschroeter commented:
rebased it on the refactored query handling

hennevogel commented on src/api/app/queries/open_requests_with_project_as_target_finder.rb:
```diff
@@ -0,0 +1,9 @@
+  end
+
+  def requests_finder
+    @relation.where(target_project: @project_name)
```
```suggestion
    @relation.where(bs_request_actions: { target_project: @project_name })
```

`@relation` is `BsRequest` and no `BsRequestAction`

hennevogel replied:
And as tests/specs don't fail it's also clear this path isn't tested. You should test it...

hennevogel committed:
Add OpenRequestsWithProjectAsTargetFinder spec

Also make the other open requests query specs a bit more precise in what they
expect.
